yellow-paper/docs/contracts/index.md

```diff
@@ -337,6 +424,16 @@ As should be clear from above, the L2 inbox doesn't need to exist for itself, it
     - Modularize the computation such that the state transitioner need not know the exact computation but merely use a separate contract as an oracle.
 - Governance/upgrade contract(s)
     - Relies on the governance/upgrade scheme being more explicitly defined
-- Explore getting rid of the specific 1:M relationship between L1 and L2
+- Explore getting rid of the specific 1:M relationship between L1 and L2 at the kernel level.
 - Forced transaction inclusion
-    - While we don't have an exact scheme, an outline was made in [hackmd](https://hackmd.io/@aztec-network/S1lRcMkvn?type=view) and the [forum](https://forum.aztec.network/t/forcing-transactions/606)
\ No newline at end of file
+    - While we don't have an exact scheme, an outline was made in [hackmd](https://hackmd.io/@aztec-network/S1lRcMkvn?type=view) and the [forum](https://forum.aztec.network/t/forcing-transactions/606)
+
+
+:::warning Comment for discussion
+In the current paradigm, any messages that are sent from the L2 contract to L1 MUST be sent to the portal address. This was to get around the access control issue of private execution and is enforced in the kernel. It practically gives us a 1:M relationship between L1 and L2, where one L1 contract can be specified as the portal for many L2, and communicate with all of them, but each L2 can only communicate with a single L1 contract.
+
+Note that while we are letting the inbox populate more values that what we did for the L1 inbox. This is more an opinionated decision than a purely technical one. Plainly speaking, we don't need to restrict the recipient of the message to a single address. We could let the contract itself figure it out. As long as the portal address exists, it CAN be used to constrain it like this.
```

Comment:
I have already included most of the outcomes I liked from that discussion. But will keep the discussion box. Something we might want to consider is also if contracts are expecting it to be the portals and it could be any immutable value the account wants, it might get some odd edge cases where people reads it as an address and specify it as a consumer etc.